### ISSUE TITLE
Error: shape does not match - Augmentations with `VideoSequential` container (works before `v0.6.9`)

### ISSUE THREAD

**Issue by elmuz** [kornia/kornia#2302]
### Crash when batch size mismatch

After upgrading from version 0.6.9 to 0.6.10 (and also confirmed in 0.6.11) I am facing the following crash due to some strange shape inconsistency. To my eyes it looks correct, and in fact it used to work. Can you help me on this?

Basically, inside my Pytorch Lightning script I have this augmentation policy, which tries to apply the same random augmentation to `faces_i`, `faces_j` and `masks_i`, for which I am extracting `_params`. I don't think this is the issue though. The code is more or less like this:
```
class MyDataLoader:
    def __init__(...)
        ...
        # Augmentation on (GPU) device
        self.geom_transform = KA.VideoSequential(
            KA.RandomHorizontalFlip(p=0.5 if multiple_identities else 0.0),
            KA.RandomAffine(
                degrees=(-10, 10),
                translate=(0.05, 0.05),
                scale=(0.95, 1.05),
                shear=(0.05, 0.05),
            ),
            data_format="BTCHW",
            same_on_frame=True,
        )

        def on_after_batch_transfer(self, batch, dataloader_idx):
            ...
            if self.trainer.training:
                faces_i = self.geom_transform(faces_i)  # => GPU/Batched data augmentation
                geom_params = self.geom_transform._params
                if self.mask:
                    masks_i = self.geom_transform(masks_i, params=geom_params)
                faces_j = self.geom_transform(faces_j, params=geom_params)
```
The issue comes at the very last batch (which has a lower size), and it complains that leading dimension (batch size) is inconsistent. To be more precise it breaks here:
```
faces_i = self.geom_transform(faces_i)  # => GPU/Batched data augmentation
```
and it raises the following error:
```
Traceback (most recent call last):
  File "src/main.py", line 16, in <module>
    LightningCLI(save_config_callback=None)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/cli.py", line 353, in __init__
    self._run_subcommand(self.subcommand)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/cli.py", line 642, in _run_subcommand
    fn(**fn_kwargs)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/call.py", line 44, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/trainer.py", line 978, in _run_stage
    self.fit_loop.run()
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/loops/fit_loop.py", line 201, in run
    self.advance()
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/loops/fit_loop.py", line 354, in advance
    self.epoch_loop.run(self._data_fetcher)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/loops/training_epoch_loop.py", line 133, in run
    self.advance(data_fetcher)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/loops/training_epoch_loop.py", line 194, in advance
    batch = call._call_strategy_hook(trainer, "batch_to_device", batch, dataloader_idx=0)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/call.py", line 288, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/strategies/strategy.py", line 261, in batch_to_device
    return model._apply_batch_transfer_handler(batch, device=device, dataloader_idx=dataloader_idx)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/core/module.py", line 325, in _apply_batch_transfer_handler
    batch = self._call_batch_hook("on_after_batch_transfer", batch, dataloader_idx)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/core/module.py", line 312, in _call_batch_hook
    return trainer_method(trainer, hook_name, *args)
  File "/usr/local/lib/python3.8/site-packages/lightning/pytorch/trainer/call.py", line 162, in _call_lightning_datamodule_hook
    return fn(*args, **kwargs)
  File "/workspace/src/wav2lip/data.py", line 463, in on_after_batch_transfer
    faces_i = self.geom_transform(faces_i, params=None)  # => GPU/Batched data augmentation
  File "/usr/local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/container/video.py", line 338, in forward
    output = self.transform_inputs(input, params, extra_args=extra_args)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/container/video.py", line 206, in transform_inputs
    input = super().transform_inputs(input, params, extra_args=extra_args)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/container/base.py", line 198, in transform_inputs
    input = InputSequentialOps.transform(input, module=module, param=param, extra_args=extra_args)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/container/ops.py", line 155, in transform
    input = module(input, params=cls.get_instance_module_param(param), **extra_args)
  File "/usr/local/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/base.py", line 212, in forward
    output = self.apply_func(in_tensor, params, flags)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/_2d/base.py", line 125, in apply_func
    trans_matrix = self.generate_transformation_matrix(in_tensor, params, flags)
  File "/usr/local/lib/python3.8/site-packages/kornia/augmentation/_2d/base.py", line 84, in generate_transformation_matrix
    trans_matrix_B = self.compute_transformation(in_tensor[to_apply], params=params, flags=flags)
IndexError: The shape of the mask [240] at index 0 does not match the shape of the indexed tensor [135, 3, 128, 128] at index 0
```
Here, `240` is the size for a normal batch, `135` is the size of last batch.

### Reproduction steps

```bash
1. Create a dataloader with `drop_last=False`
2. Apply augmentation in `on_after_batch_transfer` Lightning callback
3. Verify that dataset length is not a multiple of batch size
```

### Expected behavior

It doesn't break.

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
- PyTorch Version (e.g., 1.0): 2.0 (but also 1.x)
- OS (e.g., Linux): Linux
- How you installed PyTorch (`conda`, `pip`, source): source but also from Nvidia docker
- Build command you used (if compiling from source): `python setup.py install`
- Python version: 3.8.16
- CUDA/cuDNN version: 11.5.1/8
- GPU models and configuration: Nvidia V100
- Any other relevant information: Kornia 0.6.9 works, version 0.6.10 and 0.6.11 doesn't.
```


### Additional context

_No response_

**Comment by johnnv1** [comment]
probably something from #2117, but should be examined more closely to find the root of the issue

**Comment by johnnv1** [comment]
Fixed in #2502
